Libraries/LibWeb/HTML/HTMLIFrameElement.cpp

```diff
@@ -97,9 +97,17 @@ void HTMLIFrameElement::post_connection()
 
         if (auto navigable = content_navigable()) {
             auto traversable = navigable->traversable_navigable();
+            // AD-HOC: Since the session history traversal queue is not actually parallel, there will be a deadlock if
+            //         we append a new child navigable creation steps to the queue while we're in execution of steps for the parent traversable.
+            //         To avoid this, we set the priority of the new child navigable steps greater than parent's ie. 1.
+            int priority = 0;
+            if (traversable->is_session_history_step_locked()) {
+                priority = 1;
+            }
```

Comment:
> Since the session history traversal queue is not actually parallel

"parallel" in specification terminology means that queue items have to be processed sequentially but (optionally) in parallel with other tasks (not initiated by SHTQ). here you break sequential processing by adding priority. I would prefer to avoid doing that since that's assumption made by many navigation spec algorithms.